lib/Wikia/src/Service/User/Auth/HeliosCookieHelper.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Wikia\Service\User\Auth;
+
+use Wikia\HTTP\Response;
+use Wikia\Service\Helios\HeliosClient;
+
+class HeliosCookieHelper implements CookieHelper {
+
+	const ACCESS_TOKEN_COOKIE_NAME = 'access_token';
+	// This is set to 6 months,(365/2)*24*60*60 = 15768000
+	const ACCESS_TOKEN_COOKIE_TTL = 15768000;
+	const COOKIE_PREFIX = '';
+
+	private $heliosClient;
+
+	/**
+	 * @Inject({Wikia\Service\Helios\HeliosClient::class})
+	 */
+	public function __construct( HeliosClient $client ) {
+		$this->heliosClient = $client;
+	}
+
+
+	/**
+	 * @param HeliosClient $client
+	 */
+	public function setHeliosClient( HeliosClient $client ) {
+		$this->heliosClient = $client;
+	}
+
+
+	/**
+	 * Set the authentication cookie using the user id. This method will retrieve a new
+	 * authentication token from Helios and then set the authentication cookie.
+	 *
+	 * Warning: this should only be called once the user has been authenticated (e.g. from a 3rd party)
+	 *
+	 * @param int $userId
+	 * @param Response $response
+	 */
+	public function setAuthenticationCookieWithUserId( $userId, Response $response ) {
```

Comment:
> In order to simplify flow I would create a getter that will always return response from main request context. What do you think about it?

Can you show me an example in the code. I'm not sure I understand what you mean.